### ISSUE TITLE
[winml eval] [P1] `eval` accepts a pre-built ONNX whose I/O signature does not match `--task`; failure surfaces late as an unrelated pipeline/dataset error

### ISSUE THREAD

**Issue by zhenchaoni** [microsoft/winml-cli#529]
**Severity**: P1  
**Command**: `winml eval`

---
**Category**: Validate Inputs Up-Front + Actionable Errors — `eval` should cross-check the ONNX graph's input names against the requested task's expected inputs at load time.

**Status**: Behavior derived from static code path inspection (`commands/eval.py`, `eval/evaluate.py`). Runtime reproduction was blocked by an in-progress merge conflict in `src/winml/modelkit/commands/eval.py` on the audit branch; the reproduction string below is the path the code will follow once the conflict is resolved.

**Repro**:
```
# 1. Export a real image-classification ONNX:
uv run winml export -m microsoft/resnet-50 -o temp\resnet50.onnx
# 2. Ask eval to run it as a text-generation task:
uv run winml eval -m temp\resnet50.onnx --model-id microsoft/resnet-50 --task text-generation --samples 5
```

**Actual** (from code-path analysis):
- `_resolve_task(cfg)` returns `"text-generation"` because `cfg.task` is non-`None`. **No validation** that the task is even a known evaluator key, and **no validation** that the task is compatible with the ONNX graph.
- `WinMLAutoModel.from_onnx("temp\\resnet50.onnx", task="text-generation")` succeeds — `from_onnx` does not inspect the graph's `input` / `output` names against the task's expected I/O contract.
- `_EVALUATOR_REGISTRY.get("text-generation")` returns the text-generation evaluator, which constructs an HF `pipeline(task="text-generation", model=<resnet-onnx>)`.
- Failure surfaces several seconds later, deep in dataset preparation or first inference, as an `IndexError`/`KeyError`/missing-`input_ids` error. The user has no signal that the *task* flag is the cause; the symptom looks like a dataset or model-load bug.

**Expected**: At ONNX load time, when the user has explicitly passed `--task <T>`, compare the ONNX graph's input names against the task's documented input contract (e.g. `image-classification` → `pixel_values`; `text-generation` → `input_ids` + `attention_mask`). On mismatch, exit 2 with one line:
```
Error: ONNX file temp\resnet50.onnx exposes inputs [pixel_values] which do not
match task 'text-generation' (expects [input_ids, attention_mask]).
Did the model export with --task image-classification?
```
Secondary fix: validate `--task` against the evaluator registry at command entry; reject unknown task strings up front instead of letting `_EVALUATOR_REGISTRY.get(...)` fail late.

**Why it matters**: A user who passes the wrong `--task` (typo, wrong recall, or copy-paste from another command) gets neither a useful error nor a hint that `--task` is the wrong knob. Worst case, an evaluator that does not strictly check input shape could *complete* against random model output and report bogus accuracy numbers. Violates R4.1 (validate inputs at entry), R4.2 (reject unsupported combinations), R3.1 (actionable errors).

**Comment by DingmaomaoBJTU** [comment]
**Triage summary**

- **Component**: Eval
- **Priority**: P1

winml eval accepts a pre-built ONNX whose I/O signature does not match --task; failure surfaces late as an unrelated pipeline/dataset error. Validate I/O signature upfront.

**Comment by DingmaomaoBJTU** [comment]
**Triage summary**

- **Component**: Eval
- **Priority**: P1

`winml eval` accepts a pre-built ONNX whose I/O signature does not match the pipeline expectation, silently producing wrong results. Next step: validate the ONNX I/O signature against the expected pipeline schema before running evaluation.

**Comment by zhenchaoni** [comment]
Fixed by #694
